sqlglot/optimizer/qualify_columns.py

```diff
@@ -373,10 +399,14 @@ def get_table(self, column_name: str) -> t.Optional[exp.Identifier]:
         if isinstance(node, exp.Subqueryable):
             while node and node.alias != table_name:
                 node = node.parent
+
         node_alias = node.args.get("alias")
         if node_alias:
             return node_alias.this
-        return exp.to_identifier(table_name)
+
+        return exp.to_identifier(
+            table_name, quoted=node.this.quoted if isinstance(node, exp.Table) else False
```

Comment:
else None, we don't want it to be always false that changes behavior right

Reply:
Thanks for catching this